influxdb3_write/src/last_cache.rs

```diff
@@ -0,0 +1,2225 @@
+    fn as_array(&self) -> ArrayRef {
+        match self {
+            CacheColumnData::I64(buf) => {
+                let mut b = Int64Builder::new();
+                buf.iter().for_each(|val| match val {
+                    Some(v) => b.append_value(*v),
+                    None => b.append_null(),
+                });
+                Arc::new(b.finish())
+            }
+            CacheColumnData::U64(buf) => {
+                let mut b = UInt64Builder::new();
+                buf.iter().for_each(|val| match val {
+                    Some(v) => b.append_value(*v),
+                    None => b.append_null(),
+                });
+                Arc::new(b.finish())
+            }
+            CacheColumnData::F64(buf) => {
+                let mut b = Float64Builder::new();
+                buf.iter().for_each(|val| match val {
+                    Some(v) => b.append_value(*v),
+                    None => b.append_null(),
+                });
+                Arc::new(b.finish())
+            }
+            CacheColumnData::String(buf) => {
+                let mut b = StringBuilder::new();
+                buf.iter().for_each(|val| match val {
+                    Some(v) => b.append_value(v),
+                    None => b.append_null(),
+                });
+                Arc::new(b.finish())
+            }
+            CacheColumnData::Bool(buf) => {
+                let mut b = BooleanBuilder::new();
+                buf.iter().for_each(|val| match val {
+                    Some(v) => b.append_value(*v),
+                    None => b.append_null(),
+                });
+                Arc::new(b.finish())
+            }
+            CacheColumnData::Tag(buf) => {
+                let mut b: GenericByteDictionaryBuilder<Int32Type, GenericStringType<i32>> =
+                    StringDictionaryBuilder::new();
+                buf.iter().for_each(|val| match val {
+                    Some(v) => b.append_value(v),
+                    None => b.append_null(),
+                });
+                Arc::new(b.finish())
+            }
+            CacheColumnData::Key(buf) => {
+                let mut b: GenericByteDictionaryBuilder<Int32Type, GenericStringType<i32>> =
+                    StringDictionaryBuilder::new();
+                buf.iter().for_each(|val| {
+                    b.append_value(val);
+                });
+                Arc::new(b.finish())
+            }
+            CacheColumnData::Time(buf) => {
+                let mut b = TimestampNanosecondBuilder::new();
+                buf.iter().for_each(|val| b.append_value(*val));
+                Arc::new(b.finish())
+            }
+        }
+    }
+}
+
+/// Newtype wrapper around a [`VecDeque`] whose values are paired with an [`Instant`] that
+/// represents the time at which each value was pushed into the buffer.
+#[derive(Debug)]
+struct ColumnBuffer<T>(VecDeque<(Instant, T)>);
```

Comment:
I think that since all the column buffers for a cache get new values pushed at the same time, a separate `VecDeque` of `Instants` could be stored, vs. having to store an `Instant` next to each value in each column buffer. Each `Instant` is 16 bytes (see [here](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=846b673b1e80b3461a3c8ed6f0a92b6c)).

Reply:
Yeah, we should only be storing a single instant for each row in the last cache.